src/main/kotlin/com/autonomousapps/model/Source.kt

```diff
@@ -120,15 +120,15 @@ data class AndroidResSource(
           // See ResSpec.detects attr usage in res file.
           mapEntry.first == "name" -> AttrRef(
             type = "attr",
-            id = id.replace('.', '_')
+            id = value.replace('.', '_')
           )
           else -> null
         }
       }
 
-      private fun Pair<String, String>.isId() = first.startsWith("@+") || second.startsWith("@id")
+      private fun Pair<String, String>.isId() = second.startsWith("@+") || second.startsWith("@id")
       private fun Pair<String, String>.isToolsAttr() = first.startsWith("tools:")
-      private fun Pair<String, String>.isDataBindingExpression() = first.startsWith("@{") && first.endsWith("}")
+      private fun Pair<String, String>.isDataBindingExpression() = second.startsWith("@{") && second.endsWith("}")
```

Comment:
Why did you change these?

Reply:
From what I understand, a databinding XML file containing this instruction `binding:text="@{`Hello`}"` will be parsed as a `Pair(first = "binding:text", second = "@{`Hello`}")`? Or did I mixed things?

Reply:
Well, first, you have me wondering if there was a typo in my earlier code, except for the fact that earlier code was backed by tests that verified that it works. And then second, this change has no obvious connection to the fix you want to make, so I think you shouldn't be doing it because it conflates things. And in general I'd prefer you not change variable names without a good reason (which you also did in this PR). Again, it conflates your fix with other changes, making it harder for me to verify that your fix does just what it says and nothing else.

Reply:
Sure, I'll revert those changes.
Can you show me which and where are these tests? I'm trying to understand how this AttrRef.from gets its values from.

Reply:
Check out `ResSpec`. E.g.,
```groovy
def "gracefully handles dataBinding expressions in res files (#gradleVersion AGP #agpVersion)"
```

Reply:
btw, if you discover I have a typo and a hidden bug, do let me know! 😅

Reply:
Thanks, I'll take a look at that test class before making this PR "ready for review".
I'm currenly trying to re-enable it on a fairly large codebase (~900 modules atm) so I might discover some things along the way.

Reply:
For me to be really comfortable merging this, I'd like a new test verifying that it resolves your issue. You could add a new spec to `ResSpec`.

Reply:
I've created an `AndroidTextQuestionMarkProject` containing a single XML layout file with the `android:text="?"` that was causing the issue. I've confirmed it produced the same NPE without the fix in https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/pull/827/commits/ca2b1f39a293cd3e7af987e17baee48431bc9804.
I tried to follow the conventions of others project classes, but I'm not really used to Groovy, so I might have done something not quite right.
Anyway, you can take a look at it when you'll have time.